src/coreclr/vm/codeman.cpp

```diff
@@ -5756,6 +5756,84 @@ int NativeUnwindInfoLookupTable::LookupUnwindInfoForMethod(DWORD RelativePc,
     return -1;
 }
 
+int HotColdMappingLookupTable::LookupMappingForMethod(PTR_ULONG pLookupTable,
+                                                      int nLookupTable,
+                                                      ULONG MethodIndex)
+{
+    CONTRACTL {
+        NOTHROW;
+        GC_NOTRIGGER;
+        SUPPORTS_DAC;
+    } CONTRACTL_END;
+
+    if (nLookupTable == 0)
+    {
+        return -1;
+    }
+
+    // Casting the lookup table's size to an int is safe:
+    // We index the RUNTIME_FUNCTION table with ints, and the lookup table
+    // contains a subset of the indices in the RUNTIME_FUNCTION table.
+    // Thus, |lookup table| <= |RUNTIME_FUNCTION table|.
+
+    int high = ((nLookupTable - 1) / 2);
```

Comment:
By copying the binary search implementation in `NativeUnwindInfoLookupTable::LookupUnwindInfoForMethod`, I made `low` and `high` inclusive bounds, so I think subtracting 1 is required here. Without this, it is possible for us to index out of bounds in the `for` loop below (since we iterate while `i <= high`).

Reply:
Got it, would be more clear to write `nLookupTable/2 - 1` so that I know you want an inclusive upper bound, but that's okay.